packages/@aws-cdk/aws-lambda/lib/lambda-version.ts

```diff
@@ -72,6 +73,18 @@ export interface VersionAttributes {
  */
 export class Version extends QualifiedFunctionBase implements IVersion {
 
+  /**
+   * @param scope The cdk scope creating this resource
+   * @param id The cdk id of this resource
+   * @param versionArn The version ARN to create this version from
+   */
+  public static fromVersionArn(scope: Construct, id: string, versionArn: string): IVersion {
+    return Version.fromVersionAttributes(scope, id, {
```

Comment:
What you did here is correct, but the VersionArn now renders to something super silly, as you can tell from the test output.

It renders to `Join(":", [Select(0, Split(":", Arn)), Select(1, Split(":", Arn)), ..., Select(6, Split(":", Arn))]) + ":" + Select(7, Split(":", Arn))`, which is just a very roundabout way of getting the same value again.

Could you do something similar to the `class Import` block defined in the function below, and obtain `functionName` and `functionArn` in a more direct way (with fewer splits)?

Reply:
I can only assume that the version ARN is the function ARN with appended `:version` and split it with typescript code. That will fail if one puts in a reference here, but that might be okay?

Otherwise I cannot assume anything about the content of `versionArn`. Therefore I used the cdk Fn helper to split and concat parts 0-6 (not 0-7), but jepp - thats kind of silly.

Reply:
@rix0rrr I think it is not a valid assumtion that the value of `versionArn` is a literal string. The case described in the unit test by itself already fails that assumtion. So maybe we should keep the silly thing? At least it splits the version from the given ARN.

Reply:
>  I think it is not a valid assumtion that the value of `versionArn` is a literal string.

No, but the passed `versionArn` could go directly into `functionArn`.

The only thing that then would need splitting is `functionName`, which can be obtained by `Fn.select(6, Fn.split(':', versionArn))`.

Reply:
I've updated the implementation in addition with a rebase.
Unfortunately I couldn't compile the actual master so I haven't run (or updated) the tests.